Combinator/Combinator.fs

```diff
@@ -169,14 +169,26 @@ module Combinator =
     let manyTill p pEnd = manyTillWithCount 0 p pEnd
 
     let manyTill1 p pEnd = manyTillWithCount 1 p pEnd
-        
-    let manyN num (parser) =                              
-        takeTillWithCount 0 (fun i -> List.length i > num) parser    >>= fun result -> 
-            if result.Length <> num then
-                failwith ("Error, attempted to match " + num.ToString() + " but only got " + result.Length.ToString())
 
-            preturn result          
-    
+
+
+    let manyN num parser state =
```

Comment:
Do we need to include the state argument? It should infer that it takes a state argument already right?  Its not the end of the world to include it but it doesn't follow the pattern that the other partial functions at the top level follow